app/core/ai_monitoring.py

```
@@ -0,0 +1,186 @@
+            )
+        """)
+
+    def record_metric(self, metric_name: str, value: float, context: Optional[Dict] = None):
+        timestamp = datetime.now()
+        context_json = json.dumps(context) if context else '{}'
+        
+        with self._lock:
+            # Record raw metric
+            self._conn.execute(
+                "INSERT INTO metrics (timestamp, metric_name, value, context) VALUES (?, ?, ?, ?)",
+                (timestamp, metric_name, value, context_json)
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)